### PR TITLE
server: add --log-dir to specify logstore dir

### PR DESCRIPTION
**Pull request by 9547**
The default logstore path is under `ioutil.TempFile`. In some cases, we need to store the searched logs in a specific place.

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -40,8 +45,13 @@ type Config struct {
 }
 
 func Default() *Config {
+	dir, err := ioutil.TempDir("", "dashboard-logs")
+	if err != nil {
+		log.Fatal("Failed to create directory for storing logs", zap.Error(err))
+	}
 	return &Config{
 		DataDir:            "/tmp/dashboard-data",
+		LogDir:             dir,
```
Comment (breeswish) [inline pkg/config/config.go:54]: Looks like we should assign `""` as default?

File: pkg/config/config.go [hunk 2]
```diff
@@ -29,6 +33,7 @@ const (
 
 type Config struct {
 	DataDir          string
+	LogDir           string
```
Comment (breeswish) [inline pkg/config/config.go:36]: This name is misleading. Actually it is the directory to store searched log results. How about calling it `TempStoreDir` and use it to replace the `ioutil.TempDir`? (remember to change the cli flag name as well)

**Review by breeswish — COMMENTED** [review]
Thanks a lot for the contribution! I have a few ideas about the name.

File: pkg/config/config.go (hunk 2 above)
Comment (9547) [reply pkg/config/config.go:36]: Sorry for the later reply, I'll change it later today.

File: pkg/config/config.go (hunk 1 above)
Comment (9547) [reply pkg/config/config.go:54]: Thanks for the advice.

**Comment by 9547** [comment]
@breeswish PTAL.

**Review by breeswish — APPROVED** [review]
Sorry for the late review! This PR looks good!

**Pull request merged**